### PR TITLE
Add null check and improve load handling

### PR DESCRIPTION
**Pull request by sclements**
Addresses an issue we're seeing with URLs that take you directly into the editor.

### CODE REVIEW

File: src/Scratch.as [hunk 1]
```diff
@@ -466,6 +468,7 @@ public class Scratch extends Sprite {
 		removeLoadProgressBox();
 		System.gc();
 		if (autostart) runtime.startGreenFlags(true);
+		loadInProgress = false;
```
Comment (cwillisf) [inline src/Scratch.as:471]: Are you confident that it's safe to set this to `false` before `stepRuntime()` calls `runtime.installNewProject()`?

Comment (sclements) [inline src/Scratch.as:471]: It shouldn't happen before. ScratchRuntime::installProject() is the only caller to projectLoaded() and installNewProject() calls installProject. Scratch's projectLoaded() and loadProjectFailed() are now the only places that loadInProgress are set to false. It's cleaner and clearer this way.

Comment (cwillisf) [inline src/Scratch.as:471]: Ah, cool. Thanks for the explanation!

**Comment by cwillisf** [comment]
:+1:

**Pull request merged**